### PR TITLE
TRUNK-6206: Global property specific privileges

### PR DESCRIPTION
**Pull request by IamMujuziMoses**
## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-6206

## Checklist: I completed these to help reviewers :)

- [ ] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [ ] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [ ] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [ ] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/main/resources/org/openmrs/liquibase/updates/liquibase-update-to-latest-2.6.x.xml [hunk 1]
```diff
@@ -133,5 +133,61 @@
...
+		<addColumn tableName="global_property">
+			<column name="delete_privilege" type="VARCHAR(255)">
+				<constraints nullable="true" />
+			</column>
+		</addColumn>
+		<addForeignKeyConstraint constraintName="global_property_delete_privilege_fk"
+								 baseTableName="global_property" baseColumnNames="delete_privilege"
+								 referencedTableName="privilege" referencedColumnNames="privilege" />
+
+	</changeSet>
```
Comment (ManojLL) [inline api/src/main/resources/org/openmrs/liquibase/updates/liquibase-update-to-latest-2.6.x.xml:191]: @dkayiwa  these changes should be in `liquibase-update-to-latest-2.7.x.xml` right?

**Comment by IamMujuziMoses** [comment]
@dkayiwa @ibacher @ManojLL  could you please review this PR

**Comment by dkayiwa** [comment]
@IamMujuziMoses can you look into these merge conflicts?

**Comment by dkayiwa** [comment]
Can you look into the build failures?

File: api/src/test/java/org/openmrs/api/context/DaemonTest.java [hunk 2]
```diff
@@ -22,23 +22,41 @@
...
+		UserService userService = Context.getUserService();
+		
+		Context.addProxyPrivilege(PrivilegeConstants.GET_USERS);
+		Context.addProxyPrivilege(PrivilegeConstants.PURGE_USERS);
+		
+		userService.purgeUser(userService.getUserByUsername("jdoe"));
+		
+		Context.removeProxyPrivilege(PrivilegeConstants.GET_USERS);
+		Context.removeProxyPrivilege(PrivilegeConstants.PURGE_USERS);
+	}
```
Comment (IamMujuziMoses) [inline api/src/test/java/org/openmrs/api/context/DaemonTest.java:59]: @dkayiwa these changes fix the build failures but they are not related to this ticket, should I create another PR for them?!

Comment (dkayiwa) [reply api/src/test/java/org/openmrs/api/context/DaemonTest.java:59]: Does this mean that the latest upstream openmrs-core does not compile successfully?

Comment (dkayiwa) [reply api/src/test/java/org/openmrs/api/context/DaemonTest.java:59]: That is not true, according to this green CI builld: https://ci.openmrs.org/browse/TRUNK-MASTER/latest

Comment (IamMujuziMoses) [reply api/src/test/java/org/openmrs/api/context/DaemonTest.java:59]: Oh, okay, my apologies then🙂

Comment (dkayiwa) [reply api/src/test/java/org/openmrs/api/context/DaemonTest.java:59]: So can we remove the changes which are not related to the ticket at hand?

Comment (IamMujuziMoses) [reply api/src/test/java/org/openmrs/api/context/DaemonTest.java:59]: @dkayiwa removing these changes will lead to build failures

Comment (dkayiwa) [reply api/src/test/java/org/openmrs/api/context/DaemonTest.java:59]: Are the build failures related to this ticket?

Comment (IamMujuziMoses) [reply api/src/test/java/org/openmrs/api/context/DaemonTest.java:59]: no they're not, okay, removing them

**Comment by dkayiwa** [comment]
@IamMujuziMoses can you look into the merge conflicts?

**Pull request merged**